resources/test/tests/unit/single_test.html

```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+  <meta charset="utf-8">
+  <meta name="timeout" content="long">
+  <script src="/resources/testharness.js"></script>
+  <script src="../../nested-testharness.js"></script>
+  <title>single_test</title>
+</head>
+<body>
+<script>
+promise_test(() => {
+  return makeTest(
+      () => {
+        setup({ single_test: true });
+        done();
+      }
+    ).then(({harness, tests}) => {
+      assert_equals(harness, 'OK');
```

Comment:
This works, but I do think just using the .ini metadata would be easier to deal with when testing what the results of a testharness.js test are. (For some internals that are inconvenient to test like that using unit tests is still reasonable.)

Reply:
(Context for others: [the `resources/test/tests/functional/` directory](https://github.com/web-platform-tests/wpt/tree/dc17f8c8e3372dd30ce98d542874843f87726e8e/resources/test/tests/functional) has a bunch of tests for the testharness.js framework, but they're not great. @foolip and I are interested in improving them, and we're starting to formalize a new method here.)

I have three reasons for preferring this approach over the .ini-file approach:

- it's easier to maintain the tests and the expected results in the same file
- you can test many related concerns in a single file (we'd need one file per test if we used the .ini-file approach)
- consistency with those tests which we can't express with an ini-file

Reply:
When it's necessary to try running a test to figure out what the exact expected values should be I find in-file expectations to be pretty frustrating, leading to a trial-and-error process of copying output from the failure logs until it matches exactly. Some of the tests in resources/test/ are like that, but not all of them. When they are, a separate file that's possible to update with a tool is nicer I think.

But let's leave this alone, it follows the local convention.